.github/workflows/github-actions.yml

```diff
@@ -62,6 +62,9 @@ jobs:
         with:
           name: legateboost-wheel
           path: dist/legateboost*.whl
+      - name: Install legateboost
+        run: |
+          python -m pip install --no-deps dist/*.whl
```

Comment:
Removing `build.sh` from `ci/build_wheel.sh` ensures that here in CI, we're driving the entire CMake build via Python packaging tools (`python -m build`, in this case). That ensures that the shared library is actually copied into the wheel, and that therefore the wheels could be installed on their own.

That's not the case today. Try downloading a recent wheel artifact, e.g. from https://github.com/rapidsai/legate-boost/actions/runs/10111717786

```shell
unzip legateboost-wheel.zip
unzip -l 
```

`lib_legateboost.so` has not been getting packaged.

<details><summary>contents (click me)</summary>

```text
Archive:  legateboost-0.1-cp312-cp312-linux_x86_64.whl
  Length      Date    Time    Name
---------  ---------- -----   ----
      604  07-26-2024 13:38   legateboost/__init__.py
     5162  07-26-2024 13:38   legateboost/callbacks.py
     2682  07-26-2024 13:38   legateboost/input_validation.py
     1148  07-26-2024 13:38   legateboost/install_info.py
    34517  07-26-2024 13:38   legateboost/legateboost.py
     3156  07-26-2024 13:38   legateboost/library.py
    10209  07-26-2024 13:38   legateboost/metrics.py
    19391  07-26-2024 13:38   legateboost/objectives.py
     4635  07-26-2024 13:38   legateboost/shapley.py
     2943  07-26-2024 13:38   legateboost/special.py
    12535  07-26-2024 13:38   legateboost/utils.py
      124  07-26-2024 13:38   legateboost/models/__init__.py
     2464  07-26-2024 13:38   legateboost/models/base_model.py
     7397  07-26-2024 13:38   legateboost/models/krr.py
     4122  07-26-2024 13:38   legateboost/models/linear.py
     4882  07-26-2024 13:38   legateboost/models/nn.py
     8768  07-26-2024 13:38   legateboost/models/tree.py
        0  07-26-2024 13:38   legateboost/test/__init__.py
     1848  07-26-2024 13:38   legateboost/test/test_callbacks.py
     6744  07-26-2024 13:38   legateboost/test/test_estimator.py
     1703  07-26-2024 13:38   legateboost/test/test_examples.py
     7510  07-26-2024 13:38   legateboost/test/test_metric.py
     3787  07-26-2024 13:38   legateboost/test/test_objective.py
     3100  07-26-2024 13:38   legateboost/test/test_shapley.py
     3895  07-26-2024 13:38   legateboost/test/test_special.py
     1845  07-26-2024 13:38   legateboost/test/test_utils.py
     7832  07-26-2024 13:38   legateboost/test/test_with_hypothesis.py
     1335  07-26-2024 13:38   legateboost/test/utils.py
        0  07-26-2024 13:38   legateboost/test/models/__init__.py
     2720  07-26-2024 13:38   legateboost/test/models/test_krr.py
     4028  07-26-2024 13:38   legateboost/test/models/test_linear.py
     2925  07-26-2024 13:38   legateboost/test/models/test_nn.py
     1420  07-26-2024 13:38   legateboost/test/models/test_tree.py
      790  07-26-2024 13:38   legateboost/test/models/utils.py
     3669  07-26-2024 13:38   legateboost-0.1.data/data/lib/cmake/legateboost_python/legateboost_python-config-version.cmake
     5276  07-26-2024 13:38   legateboost-0.1.data/data/lib/cmake/legateboost_python/legateboost_python-config.cmake
     1312  07-26-2024 13:38   legateboost-0.1.data/data/lib/cmake/legateboost_python/legateboost_python-dependencies.cmake
     4182  07-26-2024 13:38   legateboost-0.1.data/data/lib/cmake/legateboost_python/legateboost_python-targets.cmake
    10174  07-26-2024 13:38   legateboost-0.1.dist-info/LICENSE
     1063  07-26-2024 13:38   legateboost-0.1.dist-info/METADATA
       99  07-26-2024 13:38   legateboost-0.1.dist-info/WHEEL
       12  07-26-2024 13:38   legateboost-0.1.dist-info/top_level.txt
     3909  07-26-2024 13:38   legateboost-0.1.dist-info/RECORD
---------                     -------
   205917                     43 files
```

</details>

Because of the use of `build.sh`, this line was always hit during the build:

https://github.com/rapidsai/legate-boost/blob/157b4271bf5f83dce5578da6b6ebe4b1d9652693/CMakeLists.txt#L17

Which I think led to it not being recompiled in the `_skbuild` directory, which I think is why it's not getitng included in wheels.